New chart: Riemann

raykrueger opened the pull request:
A new chart for [Riemann](http://riemann.io). Riemann is an event stream processor for monitoring distributed systems.

This chart simply contains the engine and nothing else. I can add the dashboard at a later point if users find this interesting. This is my first helm chart so please let me know if I've gone off the rails anywhere.

raykrueger committed:
Chart details for Riemann

Riemann is a stream based metric processing engine.

raykrueger committed:
First draft of the README

raykrueger committed:
Move the riemann website link

raykrueger committed:
Added a livenessProbe

raykrueger committed:
Remove trailing whitespace

unguiculus commented on incubator/riemann/templates/deployment.yaml:
```diff
@@ -0,0 +1,39 @@
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}"
+        imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+        {{- if .Values.service.ports.tcp.internal }}
```
Probes should not depend on the service. In fact, your service probably won't even work because the container does not expose any ports.

raykrueger replied:
Thanks! I got cute about using the values here, you're right. Those should be container ports, not service ports.

Also, the ports are already exposed via the Dockerfile in the image, which is why this works. Is it expected to have containerPort specified in any case? Easy to add.

raykrueger replied:
(I added the explicit container ports in 0a5da04)

unguiculus replied:
I think it'd be better if the container exposed ports. You could use named ports and reference them.

Here's an example:
https://github.com/kubernetes/charts/blob/master/stable/mailhog/templates/deployment.yaml#L39-L52
https://github.com/kubernetes/charts/blob/master/stable/mailhog/templates/service.yaml#L28-L42

raykrueger replied:
I did add the explicit containerPorts.

However, I'm not sure I understand the "named ports" comment. Are you referring to IANA Well-Known ports? Or is there some means of naming and referencing ports in Kubernetes? I'm finding some documentation that seems to hint at that being true. If I name a container port I can reference it by name in the service targetPort?

unguiculus commented on incubator/riemann/templates/_helpers.tpl:
```diff
@@ -0,0 +1,16 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
```
We've started namespacing templates. Please see #1785 and apply accordingly.

raykrueger replied:
Ok, thanks for the feedback. This bit came from `helm create` in version 2.6.2. I just checked 2.7.0 and it produces the same output. If there isn't a PR for that, I'll look into it later.

raykrueger replied:
Disregard the commentary on `helm create`. I changed the version linking locally and forgot. helm 2.7.0 outputs properly namespaced template helpers.

unguiculus reviewed (changes requested):
Thanks for the PR.

raykrueger committed:
Namespace the helper templates.

Feedback from https://github.com/kubernetes/charts/pull/2652

raykrueger committed:
Probes should use exposed container port values.

Oops. I got too cute using .Values here and was using the service
values. The liveness and readiness probes must be pointed at exposed
container ports.

Port 5555 is exposed in the image via `EXPOSE 5555`, but I have also
explicitly exposed it via a containers.ports.containerPort entry.

unguiculus commented on incubator/riemann/templates/configmap.yaml:
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-configmap
```
Should be: `name: {{ template "riemann.fullname" . }}`

unguiculus replied:
Add standard labels.

raykrueger replied:
I've renamed the configMap to use the template and drop the -configmap suffix.
I've added the missing standard labels to the configMap (and the pod defined in the deployment)

unguiculus commented on incubator/riemann/values.yaml:
```diff
@@ -0,0 +1,44 @@
+# does not support horizontal scaling out of the box.
+replicaCount: 1
+image:
+  repository: raykrueger/riemann:0.2.14
```
Separate repo and tag.

```yaml
image:
  repository: raykrueger/riemann
  tag: 0.2.14
```

raykrueger replied:
I wasn't seeing a lot of consistency here among the stable images, so I went with the simplest form. Is there tooling that expects separate values?

raykrueger committed:
Rename the configMap to follow standard naming.

raykrueger committed:
Added missing standard labels

raykrueger committed:
Separate the image and tag values

raykrueger committed:
Switch to named ports for service/pod communication

Naming the container ports and referencing those in the service is much
easier and cleaner.

raykrueger commented:
@unguiculus Ready for you to have another look when you can. Thanks!

unguiculus commented on incubator/riemann/Chart.yaml:
```diff
@@ -0,0 +1,16 @@
+sources:
+- https://github.com/raykrueger/riemann-docker
+maintainers:
+- name: Ray Krueger
```
Use Github username.

unguiculus commented on incubator/riemann/templates/service.yaml:
```diff
@@ -0,0 +1,33 @@
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+      {{- if .Values.service.ports.tcp }}
```
I find it kind of weird that you indent the if statement instead of the code that is wrapped inside of it. Please do it the other way around. Apply everywhere.

unguiculus commented:
/ok-to-test

raykrueger committed:
Use my github name

raykrueger committed:
Fix formatting

If lines were indented weird. New Output:

    spec:
      type: ClusterIP
      ports:
      - name: riemann-tcp
        port: 5555
        targetPort: riemann-tcp
        protocol: TCP
      - name: riemann-udp
        port: 5555
        targetPort: riemann-udp
        protocol: UDP
      - name: riemann-ws
        port: 5556
        targetPort: riemann-ws
        protocol: TCP

unguiculus commented:
/lgtm
